components/services/bat_ledger/bat_ledger_impl.cc

```diff
@@ -594,6 +594,16 @@ void BatLedgerImpl::UpdateMediaDuration(
   ledger_->UpdateMediaDuration(window_id, publisher_key, duration, first_visit);
 }
 
+void BatLedgerImpl::IsPublisherRegistered(
+    const std::string& publisher_id,
+    IsPublisherRegisteredCallback callback) {
+  ledger_->IsPublisherRegistered(
+      publisher_id, [shared_callback = std::make_shared<decltype(callback)>(
```

Comment:
It's a simpler way to wrap a callback with an `std::function` for use by `LedgerImpl` than using `CallbackHolder` plus a member function callback. The `std::shared_ptr` is just a way of putting the callback into a copyable `std::function`.

Reply:
well, if this is a new method/callback, why use std::function at all? This stuff was all just for backwards compat and pre-brave-core ios

Reply:
The main reason we have CallbackHolder instead of a simpler wrapper is to handle lifetime issues by using WeakPtr. Since we're dealing with mojo I'm not sure if that was actually a valid concern or not, but to be safe I would continue using it for anything that has to be std::function

Reply:
Although I dislike adding anything to the `Ledger` interface, I needed this new method in order to efficiently show the publisher checkmark. The options are:

- Add a new interface, separate from `Ledger`.
  - This will push this PR out quite a bit while we design new interfaces. We'd probably have to merge some of that work before this one.
- Use `base::OnceCallback` directly in the `Ledger` interface.
  - This will just push the `shared_ptr` creation into `LedgerImpl` since we'll have to create an `std::function` there which wraps the `OnceCallback`.
  - Also, it will break the existing consistency (unfortunate as it is).
- Just use `CallbackHolder` here.
  - Unfortunately, `CallbackHolder` can leak memory. I can continue to use the pattern, though.

My preference would be to use the lambda as-is, but I'm also fine with using `CallbackHolder` if that's preferred. I would prefer not mess with the `Ledger` interface style in this PR, though.

Reply:
Updated to use `CallbackHolder`.